ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                    size_t nBytes = 4;
+                    for (nIRecord = 0;
+                         nIRecord <
+                         phMiraMonLayer->pMultRecordIndex[nIElem].nMR;
+                         nIRecord++)
+                    {
+                        GoToFieldOfMultipleRecord(nIElem, nIRecord, nIField);
+
+                        fread_function(phMiraMonLayer->szStringToOperate +
+                                           nBytes,
+                                       phMiraMonLayer->pMMBDXP->pField[nIField]
+                                           .BytesPerField,
+                                       1, phMiraMonLayer->pMMBDXP->pfDataBase);
+                        (phMiraMonLayer->szStringToOperate +
+                         nBytes)[phMiraMonLayer->pMMBDXP->pField[nIField]
+                                     .BytesPerField] = '\0';
+                        MM_RemoveLeadingWhitespaceOfString(
+                            phMiraMonLayer->szStringToOperate + nBytes);
+                        MM_RemoveWhitespacesFromEndOfString(
+                            phMiraMonLayer->szStringToOperate);
+
+                        nBytes +=
+                            strlen(phMiraMonLayer->szStringToOperate + nBytes);
+                        if (phMiraMonLayer->pMMBDXP->CharSet ==
+                            MM_JOC_CARAC_OEM850_DBASE)
+                            MM_oemansi_n(
+                                phMiraMonLayer->szStringToOperate + nBytes,
+                                phMiraMonLayer->pMMBDXP->pField[nIField]
+                                    .BytesPerField);
+
+                        if (phMiraMonLayer->pMMBDXP->CharSet !=
+                            MM_JOC_CARAC_UTF8_DBF)
+                        {
+                            // MiraMon encoding is ISO 8859-1 (Latin1) -> Recode to UTF-8
+                            char *pszString = CPLRecode(
+                                phMiraMonLayer->szStringToOperate + nBytes,
+                                CPL_ENC_ISO8859_1, CPL_ENC_UTF8);
+
+                            CPLStrlcpy(
+                                phMiraMonLayer->szStringToOperate + nBytes,
+                                pszString,
+                                (size_t)phMiraMonLayer->pMMBDXP->pField[nIField]
+                                        .BytesPerField +
+                                    1);
+
+                            CPLFree(pszString);
+                        }
+
+                        if (nIRecord <
+                            phMiraMonLayer->pMultRecordIndex[nIElem].nMR - 1)
+                        {
+                            strcat(phMiraMonLayer->szStringToOperate + nBytes,
+                                   ",");
+                            nBytes += 1;
+                        }
+                        else
+                        {
+                            strcat(phMiraMonLayer->szStringToOperate + nBytes,
```

Comment:
cf previous remark. A Json array should just end with a square bracket ]

Reply:
Done